packages/ui/atoms/ScrollReveal/ScrollReveal.js

```diff
@@ -0,0 +1,56 @@
+import { withMountWhenInView } from '@studiometa/js-toolkit';
+import { Transition } from '../../primitives/index.js';
+
+/**
+ * @typedef {ScrollReveal & {
+ *   $refs: {
+ *     target?: HTMLElement,
+ *   },
+ * }} ScrollRevealInterface
+ */
+
+/**
+ * ScrollReveal class.
+ */
+export default class ScrollReveal extends withMountWhenInView(Transition) {
+  /**
+   * Config.
+   */
+  static config = {
+    ...Transition.config,
+    name: 'ScrollReveal',
+    refs: ['target'],
+    options: {
+      ...Transition.config.options,
+      enterKeep: {
+        type: Boolean,
+        default: true,
+      },
+      intersectionObserver: {
+        type: Object,
+        default: () => ({ threshold: [0, 1] }),
+      },
+    },
+  };
+
+  /**
+   * Get the transition target.
+   *
+   * @this {ScrollRevealInterface}
+   * @returns {HTMLElement}
+   */
+  get target() {
+    return this.$refs.target ?? this.$el;
```

Comment:
**question**: great to fallback on `$el`. Wondering if the component will throw a warning `the target ref is missing for ScrollReveal component` when used on `$el` ?

Reply:
The warnings are only displayed in dev mode since the [v2.3.0](https://github.com/studiometa/js-toolkit/releases/tag/2.3.0) of @studiometa/js-toolkit (see [src](https://github.com/studiometa/js-toolkit/blob/52712abf64229c0e8b072664ee7cee9318427ef8/packages/js-toolkit/Base/managers/RefsManager.js#L63-L70)). 

So this is not blocking and will not impact production envs 😉